packages/sit-onyx/src/utils/forms.ts

```diff
@@ -1,3 +1,23 @@
+const getValidityStateProperties = () =>
+  Object.entries(Object.getOwnPropertyDescriptors(ValidityState.prototype))
+    .filter(([_, value]) => value.enumerable)
+    .map(([key]) => key) as (keyof ValidityState)[];
+
+/**
+ * The standard HTML ValidityState is an object with getter properties.
+ * This makes it hard to compare changes. With below function, we iterate through all
+ * getters and transform them to properties of a new blanc object.
+ */
+export const transformValidityStateToObject = (validityState: ValidityState): ValidityState => {
+  return getValidityStateProperties().reduce<Record<keyof ValidityState, boolean>>(
```

Comment:
```suggestion
  return getValidityStateProperties().reduce(
```

Does this work? Since the default value of the reducer is already type casted, it should be the same